### PR TITLE
Add drug and substance structured data to site

### PR DESCRIPTION
**Pull request by craiga**
### Add drug and substance structured data to site

[Airtable ticket](https://airtable.com/tbl7CISHKkr8Kdeh2/viwlDIU6cp8cbQYhj/rec30Osl21ceHh9y9?blocks=hide)

[GitHub issue](https://github.com/MHRA/products/issues/371)

giphy-17

### Acceptance Criteria

- [x] Microdata should be included as JSON-LD.
- [x] Substances are marked up as a Substance  (http://schema.org/Substance)
- [x] Drug search results are marked up as drugs (http://schema.org/Drug).
- [x] Lists of these things are marked up as lists (https://schema.org/ItemList).

### Testing information

Structured data can be viewed using [Google's Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool/u/0/).

### Pre-flight Checklist

- [ ] Tests
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [x] Branch test approved

### CODE REVIEW

File: medicines/web/components/mip/index.tsx [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import moment from 'moment';
 import { useRouter } from 'next/router';
+import { stringify } from 'querystring';
 import React, { FormEvent, useEffect } from 'react';
 import ReactGA from 'react-ga-gtm';
+import { JsonLd } from 'react-schemaorg';
```
Comment (pataruco) [inline medicines/web/components/mip/index.tsx]: I did not know about this library 💥
This is very nice!

**Comment by rjkerrison** [comment]
Tested on branch.

On an index page, there's an ItemList of Substance. This appears to be correct behaviour.
On a substance page, there's only one entry, and it is the first item and it is of type Substance. Expected an ItemList of all entries, each entry being a Drug.

File: medicines/web/src/pages/substance-index/index.tsx [hunk 2]
```diff
@@ -49,6 +51,25 @@ const App: NextPage = () => {
...
+                  '@type': 'Substance',
+                  name: substance.name,
+                  url:
+                    'https://products.mhra.gov.uk/substance?query=' +
+                    encodeURIComponent(substance.name),
+                },
+              };
+            }),
+          }}
+        />
```
Comment (TimboTambo) [inline medicines/web/src/pages/substance-index/index.tsx]: May be worth extracting this to a separate, self-contained component in the same file to make the contents of this component cleaner.

Comment (craiga) [reply medicines/web/src/pages/substance-index/index.tsx]: Sounds like a good idea.

**Pull request merged**